src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -239,6 +243,45 @@ private static void createBranches(LfNetwork lfNetwork, LoadingContext loadingCo
                 legNumber++;
             }
         }
+
+        List<Set<LfBus>> connectedSets = getNonImpedantConnectedSets(lfNetwork);
+        connectedSets.stream()
+            .filter(set -> set.size() > 2) // Two LfBuses with one voltage control is never a problem
+            .forEach(set -> set.forEach(b -> removeConflictuousDiscreteVoltageControlled(b, set)));
+    }
+
+    private static List<Set<LfBus>> getNonImpedantConnectedSets(LfNetwork network) {
+        List<LfBranch> nonImpedantBranches = network.getBranches().stream()
+            .filter(LfNetworkLoaderImpl::isNonImpedantBranch)
+            .filter(b -> b.getBus1() != null && b.getBus2() != null)
+            .collect(Collectors.toList());
+
+        Graph<LfBus, LfBranch> nonImpedantSubGraph = new Pseudograph<>(LfBranch.class);
+        for (LfBranch branch : nonImpedantBranches) {
+            nonImpedantSubGraph.addVertex(branch.getBus1());
+            nonImpedantSubGraph.addVertex(branch.getBus2());
+            nonImpedantSubGraph.addEdge(branch.getBus1(), branch.getBus2(), branch);
+        }
+
+        return new ConnectivityInspector<>(nonImpedantSubGraph).connectedSets();
+    }
+
+    private static boolean isNonImpedantBranch(LfBranch branch) {
+        PiModel piModel = branch.getPiModel();
+        return piModel.getZ() < DcEquationSystem.LOW_IMPEDANCE_THRESHOLD;
+    }
+
+    private static void removeConflictuousDiscreteVoltageControlled(LfBus lfBus, Set<LfBus> nonImpedantSet) {
```

Comment:
What we want:
- Check if in a non impedant connected set we have several controlled bus (later, we will have to manage controlled buses by continuous voltage control or discrete voltage controls) ; 
- If yes, only one voltage control is kept, the other one will be removed ;
- All the controllers of the removed voltage control are transferred in the first voltage control.